sweep/fee_bumper.go

```diff
@@ -921,3 +928,235 @@ func calcCurrentConfTarget(currentHeight, deadline int32) uint32 {
+		len(inputs))
+
+	return sweepTx, txFee, nil
+}
+
+// prepareSweepTx returns the tx fee, an optional change amount and an optional
+// locktime after a series of validations:
+// 1. check the locktime has been reached.
+// 2. check the locktimes are the same.
+// 3. check the inputs cover the outputs.
+//
+// NOTE: if the change amount is below dust, it will be added to the tx fee.
+func prepareSweepTx(inputs []input.Input, changePkScript []byte,
+	feeRate chainfee.SatPerKWeight, currentHeight int32) (
+	btcutil.Amount, fn.Option[btcutil.Amount], fn.Option[int32], error) {
+
+	noChange := fn.None[btcutil.Amount]()
+	noLocktime := fn.None[int32]()
+
+	// Creating a weight estimator with nil outputs and zero max fee rate.
+	// We don't allow adding customized outputs in the sweeping tx, and the
+	// fee rate is already being managed before we get here.
+	inputs, estimator, err := getWeightEstimate(
+		inputs, nil, feeRate, 0, changePkScript,
+	)
+	if err != nil {
+		return 0, noChange, noLocktime, err
+	}
+
+	txFee := estimator.fee()
+
+	var (
+		// Track whether any of the inputs require a certain locktime.
+		locktime = int32(-1)
+
+		// We keep track of total input amount, and required output
+		// amount to use for calculating the change amount below.
+		totalInput     btcutil.Amount
+		requiredOutput btcutil.Amount
+	)
+
+	// Go through each input and check if the required lock times have
+	// reached and are the same.
+	for _, o := range inputs {
+		// If the input has a required output, we'll add it to the
+		// required output amount.
+		if o.RequiredTxOut() != nil {
+			requiredOutput += btcutil.Amount(
+				o.RequiredTxOut().Value,
+			)
+		}
+
+		// Update the total input amount.
+		totalInput += btcutil.Amount(o.SignDesc().Output.Value)
+
+		lt, ok := o.RequiredLockTime()
+
+		// Skip if the input doesn't require a lock time.
+		if !ok {
+			continue
+		}
+
+		// Check if the lock time has reached
+		if lt > uint32(currentHeight) {
+			return 0, noChange, noLocktime, ErrLocktimeImmature
+		}
+
+		// If another input commits to a different locktime, they
+		// cannot be combined in the same transaction.
+		if locktime != -1 && locktime != int32(lt) {
+			return 0, noChange, noLocktime, ErrLocktimeConflict
+		}
+
+		// Update the locktime for next iteration.
+		locktime = int32(lt)
+	}
+
+	// Make sure total output amount is less than total input amount.
+	if requiredOutput+txFee > totalInput {
+		return 0, noChange, noLocktime, fmt.Errorf("insufficient "+
+			"input to create sweep tx: input_sum=%v, "+
+			"output_sum=%v", totalInput, requiredOutput+txFee)
+	}
+
+	// The value remaining after the required output and fees is the
+	// change output.
+	changeAmt := totalInput - requiredOutput - txFee
+	changeAmtOpt := fn.Some(changeAmt)
+
+	// We'll calculate the dust limit for the given changePkScript since it
+	// is variable.
+	changeLimit := lnwallet.DustLimitForSize(len(changePkScript))
```

Comment:
suggestion: `changeFloor` to imply it's a lower limit. Alternatively `changeMinimum`

Reply:
yeah more descriptive, changed.